examples/cern_app.py

```diff
@@ -0,0 +1,157 @@
+
+2. Ensure you have ``gunicorn`` package installed:
+
+   .. code-block:: console
+
+      cdvirtualenv src/invenio-oauthclient
+      pip install -e .[https]
+
+
+3. Ensure you have ``openssl`` installed in your system (Most of the Linux
+   distributions has it by default.).
+
+
+3. Grab the *client_id* and *secret_uri* after registering the application
+   and add them to your instance configuration as `consumer_key` and
+   `consumer_secret`.
+
+   .. code-block:: console
+
+       $ export CERN_APP_CREDENTIALS_KEY=my_cern_client_id
+       $ export CERN_APP_CREDENTIALS_SECRET=my_cern_secret_uri
+
+4. Create database and tables:
+
+   .. code-block:: console
+
+       $ cd examples
+       $ flask -a cern_app.py db init
+       $ flask -a cern_app.py db create
+
+You can find the database in `examples/cern_app.db`.
+
+5. Create the key and the certificate in order to run a HTTPS server:
+
+   .. code-block:: console
+
+       $ openssl genrsa 1024 > ssl.key
+       $ openssl req -new -x509 -nodes -sha1 -days 365 -key ssl.key > ssl.crt
+
+6. Run gunicorn server:
+
+   .. code-block:: console
+
+       $ gunicorn -b :5000 --certfile=ssl.crt --keyfile=ssl.key cern_app:app
+
+7. Open in a browser the page `https://localhost:5000/`.
+
+   You will be redirected to CERN to authorize the application.
+
+   Click on `Grant` and you will be redirected back to
+   `http://localhost:5000/oauth/signup/cern/`
+
+   Now, you will be again in homepage but this time it say: `hello youremail@cern.ch`.
+
+   You have completed the user authorization.
+"""
+
+from __future__ import absolute_import, print_function
+
+import copy
+import os
+
+from flask import Flask, redirect, url_for
+from flask_babelex import Babel
+from flask_cli import FlaskCLI
+from flask_login import current_user
+from flask_menu import Menu as FlaskMenu
+from flask_oauthlib.client import OAuth as FlaskOAuth
+from invenio_accounts import InvenioAccounts
+from invenio_accounts.views import blueprint as blueprint_user
+from invenio_db import InvenioDB
+
+from invenio_oauthclient import InvenioOAuthClient
+from invenio_oauthclient.contrib import cern
+from invenio_oauthclient.views.client import blueprint as blueprint_client
+from invenio_oauthclient.views.settings import blueprint as blueprint_settings
+
+
+# [ Configure application credentials ]
+CERN_APP_CREDENTIALS = dict(
+    consumer_key=os.environ.get('CERN_APP_CREDENTIALS_KEY'),
+    consumer_secret=os.environ.get('CERN_APP_CREDENTIALS_SECRET'),
+)
+
+# Create Flask application
+app = Flask(__name__)
+
+CERN_REMOTE_APP = copy.deepcopy(cern.REMOTE_APP)
+CERN_REMOTE_APP["params"].update(dict(request_token_params={
+    "scope": "Name Email Bio Groups",
+}))
+
+app.config.update(
+    SQLALCHEMY_DATABASE_URI=os.environ.get(
+        'SQLALCHEMY_DATABASE_URI', 'sqlite:///cern_app.db'
+    ),
+    OAUTHCLIENT_REMOTE_APPS=dict(
+        cern=CERN_REMOTE_APP
+    ),
+    CERN_APP_CREDENTIALS=CERN_APP_CREDENTIALS,
+    DEBUG=True,
```

Comment:
@jirikuncar But at the end, it is not using the flask built-in server, it is using ``gunicorn`` so this is the easiest way to set the debug mode.